### ISSUE TITLE
install errors

### ISSUE THREAD

**Issue by biozzq** [lczech/grenedalf#11]
Hi @lczech 
Very nice tool. When installing this tool, i get the following errors. If possible, could you provide a binary program or any advice on installing under anaconda environment?
```
[ 21%] Building CXX object libs/genesis/lib/genesis/CMakeFiles/genesis_lib_obj.dir/__/__/__/__/genesis_unity_sources/lib/all.cpp.o
/public/home/zhengzhuqing/01.software/install/grenedalf/build/genesis_unity_sources/lib/all.cpp: In member function ‘void genesis::population::BedReader::read_(std::shared_ptr<genesis::utils::BaseInputSource>, std::function<void(genesis::population::BedReader::Feature&&)>) const’:
/public/home/zhengzhuqing/01.software/install/grenedalf/build/genesis_unity_sources/lib/all.cpp:7696:24 warning: missing initializer for member ‘genesis::population::BedReader::Feature::chrom’ [-Wmissing-field-initializers]
         feat = Feature{};

make[3]: *** [libs/genesis/lib/genesis/CMakeFiles/genesis_lib_obj.dir/__/__/__/__/genesis_unity_sources/lib/all.cpp.o] Error 1
make[2]: *** [libs/genesis/lib/genesis/CMakeFiles/genesis_lib_obj.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [build] Error 2
```

Sincerely,
Zheng zhuqing

**Comment by lczech** [comment]
Hi @biozzq,

thanks for reporting this, that is new, and seems to be just an overly cautious compiler... Which compiler are you using? Could you maybe append the whole cmake output here?

I'm pushing a fix to the `dev` branch here now. To get that, please use the green "Code" button on the dev branch [here](https://github.com/lczech/grenedalf/tree/dev), and compile that. (You could try with other methods, but updating git submodules etc is a bit tricky then...)

The `dev` branch currently has a few updates to the command line interface that are not yet reflected in the wiki, so you might need to use the `--help` of each command to get the exact options they support. Not too many changes, but might be helpful.

Let me know if that fixes it.
Cheers and so long
Lucas

**Comment by biozzq** [comment]
Dear Lucas

Thanks for your help. I used the green "code" button to download the master branch and compiled that, the same error still persists. I have attached the full output of the make process, hopefully this will give you more information. I noticed that it might be the htslib causing the error, if so I could make it with a local folder containing htslib. How to specific the local folder containing htslib when making?

Best regards,
Zheng zhuqing

**Comment by lczech** [comment]
Hey Zheng zhuqing,

yep, the `master` branch does not have that fix yet - as I mentioned, you need the `dev` branch for now, from [here](https://github.com/lczech/grenedalf/tree/dev) for instance.

I also cannot see your attached output, can you please try again? The error does very likely not come from htslib, so I wouldn't bother trying to build that on your own.

Cheers
Lucas

**Comment by biozzq** [comment]
Sorry, I forgot to attach the output file. 
[log.txt](https://github.com/lczech/grenedalf/files/12557005/log.txt)

**Comment by lczech** [comment]
Ah okay, thanks, that worked, and there I see the issue. You are using GCC 4.8.5 to compile, which is too old. That version is 8 years old, and does not even fully support C++11 yet (from 12 years ago). Please try with a newer compiler. It might work with GCC 4.9 already, but as stated in the [readme](https://github.com/lczech/grenedalf), we only test with GCC >= 7 at the moment, so use any more recent one to be on the safe side.

**Comment by biozzq** [comment]
Thank you, yes, I am working on a cluster. I will try a more recent compiler.

**Comment by biozzq** [comment]
Dear Lucas
I have updated the complier using conda and downloaded the `dev` branch, but I get another error (attached here). 
```
conda create -p /public/home/zhengzhuqing/01.software/gcc -c conda-forge gcc
conda activate /public/home/zhengzhuqing/01.software/gcc
conda install -c anaconda cmake
conda install -c conda-forge cxx-compiler
```
[log.txt](https://github.com/lczech/grenedalf/files/12557696/log.txt)

Best regards,
Zheng zhuqing

**Comment by lczech** [comment]
Haha oh wow, now you've went with a compiler that is newer than anything I used for testing so far (but I've added it now as well to the tests) - and that compiler somehow defined some constants differently from all previous versions. Sometimes, these subtle changes are really annoying, and I needed to rewrite the involved code in a completely different way to not depend on those weird constants... Thanks for spotting this! (I would have run into this anyway eventually, so good to fix it now already)

Anyway, long story short, should be fixed on the `dev` branch now. Please try again, and thanks for your patience!

Cheers
Lucas

**Comment by biozzq** [comment]
Thanks for your work. The latest `dev` branch worked for me, although there are still some warnings.
```
configure: WARNING: GCS support not enabled: requires libcurl support
configure: WARNING: S3 support not enabled: requires libcurl support
checking for library containing regcomp... none required
checking whether PTHREAD_MUTEX_RECURSIVE is declared... yes
configure: creating ./config.status
config.status: creating config.mk
config.status: creating htslib.pc.tmp
config.status: creating config.h
config.status: linking htscodecs_bundled.mk to htscodecs.mk
[ 14%] Performing build step for 'htslib'
cram/cram_index.c: In function 'cram_index_build':
cram/cram_index.c:743:17: warning: variable 'hpos' set but not used [-Wunused-but-set-variable]
  743 |     off_t cpos, hpos;
      |                 ^~~~
In file included from htscodecs/htscodecs/rANS_static4x16pr.c:62:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr_avx2.c:48:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr_avx512.c:54:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr_sse4.c:47:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr.c:45:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
cram/cram_index.c: In function 'cram_index_build':
cram/cram_index.c:743:17: warning: variable 'hpos' set but not used [-Wunused-but-set-variable]
  743 |     off_t cpos, hpos;
      |                 ^~~~
In file included from htscodecs/htscodecs/rANS_static4x16pr.c:62:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr_avx2.c:48:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr_avx512.c:54:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr_sse4.c:47:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
In file included from htscodecs/htscodecs/rANS_static32x16pr.c:45:
htscodecs/htscodecs/rANS_static16_int.h: In function 'encode_freq1':
htscodecs/htscodecs/rANS_static16_int.h:316:9: warning: variable 'tab_size' set but not used [-Wunused-but-set-variable]
  316 |     int tab_size = 0, i, j, z;
      |         ^~~~~~~~
test/test-vcf-api.c: In function 'bcf_to_vcf':
test/test-vcf-api.c:361:9: warning: ignoring return value of 'fwrite' declared with attribute 'warn_unused_result' [-Wunused-result]
  361 |         fwrite(line.s,1,line.l,stdout);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 17%] Performing install step for 'htslib'
[ 19%] Completed 'htslib'
[ 19%] Built target htslib
[ 21%] Building CXX object libs/genesis/lib/genesis/CMakeFiles/genesis_lib_obj.dir/__/__/__/__/genesis_unity_sources/lib/all.cpp.o
[ 21%] Built target genesis_lib_obj
[ 24%] Linking CXX static library /public/home/zhengzhuqing/01.software/install/grenedalf-dev/libs/genesis/bin/libgenesis.a
[ 24%] Built target genesis_lib_static
[ 26%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/analyze/diversity.cpp.o
[ 29%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/analyze/fst.cpp.o
[ 31%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/convert/frequency.cpp.o
[ 34%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/convert/sync.cpp.o
[ 36%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/simulate/simulate.cpp.o
[ 39%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/tools/citation.cpp.o
[ 41%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/tools/license.cpp.o
[ 43%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/tools/version.cpp.o
[ 46%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/tools/wiki.cpp.o
[ 48%] Building CXX object CMakeFiles/grenedalf.dir/src/commands/visualize/afs_heatmap.cpp.o
[ 51%] Building CXX object CMakeFiles/grenedalf.dir/src/main.cpp.o
[ 53%] Building CXX object CMakeFiles/grenedalf.dir/src/options/file_input.cpp.o
[ 56%] Building CXX object CMakeFiles/grenedalf.dir/src/options/file_output.cpp.o
[ 58%] Building CXX object CMakeFiles/grenedalf.dir/src/options/global.cpp.o
[ 60%] Building CXX object CMakeFiles/grenedalf.dir/src/options/poolsizes.cpp.o
[ 63%] Building CXX object CMakeFiles/grenedalf.dir/src/options/table_output.cpp.o
[ 65%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_file_frequency_table.cpp.o
[ 68%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_file_pileup.cpp.o
[ 70%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_file_sam.cpp.o
[ 73%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_file_sync.cpp.o
[ 75%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_file_vcf.cpp.o
[ 78%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_filter_numerical.cpp.o
[ 80%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_filter_region.cpp.o
[ 82%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_input.cpp.o
[ 85%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_reference_genome.cpp.o
[ 87%] Building CXX object CMakeFiles/grenedalf.dir/src/options/variant_sample_names.cpp.o
[ 90%] Building CXX object CMakeFiles/grenedalf.dir/src/options/window.cpp.o
[ 92%] Building CXX object CMakeFiles/grenedalf.dir/src/tools/cli_setup.cpp.o
[ 95%] Building CXX object CMakeFiles/grenedalf.dir/src/tools/misc.cpp.o
[ 97%] Building CXX object CMakeFiles/grenedalf.dir/src/tools/references.cpp.o
[100%] Linking CXX executable /public/home/zhengzhuqing/01.software/install/grenedalf-dev/bin/grenedalf

[100%] Built target grenedalf
Done.
```

Cheers,
Zheng zhuqing

**Comment by lczech** [comment]
Perfect, happy to hear, and going to close this issue then. If you have any more trouble, feel free to open another one :-)

As for those warnings: Yes, I had them in my tests with GCC 13 as well, and tried to silence them, but couldn't get it set up properly in an easy way. They do not come from code that I wrote, and hence I have not much control over that. I have opened an [issue](https://github.com/samtools/htscodecs/issues/93) in the library that is causing the warnings, and hope this will be fixed at some point. Either way, it is absolutely nothing to worry about - just warning that a particular variable is there, but never used. Not a big deal.
